### PR TITLE
[BOSS-440] Detect and rotate upside down 360 panorama results.

### PR DESCRIPTION
**Pull request by bluecamel**

### CODE REVIEW

File: stitcher/include/airmap/stitcher.h [hunk 1]
```diff
@@ -125,15 +125,16 @@ class OperationsMonitoredStitcher : public MonitoredStitcher
...
+        std::shared_ptr<Camera> camera, UpdatedCb updatedCb = []() {})
+        : MonitoredStitcher(OperationsEstimator::create(
+              camera, logger, updatedCb, parameters.enableEstimate,
+              parameters.enableEstimateLog))
         , _camera(camera)
-        , _monitor(Monitor::create(_estimator, logger, parameters.enableElapsedTime,
-                                   parameters.enableEstimateLog))
+        , _monitor(Monitor::create(_estimator, logger,
+                                   parameters.enableElapsedTime,
+                                   parameters.enableElapsedTimeLog))
```
Comment (bluecamel) [inline stitcher/include/airmap/stitcher.h:137]: This was unrelated, but while testing I noticed that the last parameter to `Monitor::create` was wrong.

File: stitcher/src/stitcher.cpp [hunk 2]
```diff
@@ -787,6 +789,152 @@ LowLevelOpenCVStitcher::prepareExposureCompensation(
...
+        float x = (warped_images_[i].size().width - 1) / 2.f;
+        float y = (warped_images_[i].size().height - 1) / 2.f;
+        cv::Point2f center(x, y);
+        cv::Mat rotation = cv::getRotationMatrix2D(center, 180., 1.);
+        cv::warpAffine(warped_images_[i], warped_rotated_image, rotation,
+                       warped_images_[i].size());
+        warped_rotated_images.push_back(warped_rotated_image);
+    }
+
+    // TODO(bkd): std::move on const reference seems to be okay?
```
Comment (bluecamel) [inline stitcher/src/stitcher.cpp:826]: @keith-bennett-airmap Can you confirm that this makes sense?  I haven't read super deep here, but I think that this is okay, if unintuitive.

Comment (remster) [reply stitcher/src/stitcher.cpp:826]: I don't understand the dilemma/comment

File: stitcher/src/opencv/matchers.cpp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+ThreeSixtyPanoramaOrientationMatcher::ThreeSixtyPanoramaOrientationMatcher(
+    bool try_use_gpu, float match_conf, int num_matches_thresh1,
+    int num_matches_thresh2)
+    : BestOf2NearestMatcher(try_use_gpu, match_conf, num_matches_thresh1,
+                            num_matches_thresh2)
+{
+}
+
+void ThreeSixtyPanoramaOrientationMatcher::operator()(
```
Comment (remster) [inline stitcher/src/opencv/matchers.cpp:53]: Quite hard to follow the code, can you document at a higher lavel what's being done here?

File: stitcher/src/stitcher.cpp (hunk 2 above)
Comment (bluecamel) [reply stitcher/src/stitcher.cpp]: The short story is that I read [this article](https://www.nextptr.com/tutorial/ta1211389378/beware-of-using-stdmove-on-a-const-lvalue) and was concerned that I might have an issue, but I think that it's fine.

File: stitcher/src/opencv/matchers.cpp (hunk 3 above)
Comment (bluecamel) [reply stitcher/src/opencv/matchers.cpp:52]: I'm very sorry.  I saw these as pretty simple modifications of OpenCV's `BestOf2NearestMatcher` and wasn't thinking about folks not being familiar with those.  I've added a [bunch of comments](https://github.com/airmap/image-processing/pull/13/commits/21afb7e42bfceaa6619db2a145a33ba2032ffb0a) that hopefully makes all of this clearer, but please let me know if not.

**Review by remster — APPROVED** [review]

File: stitcher/src/stitcher.cpp (hunk 2 above)
Comment (keith-bennett-airmap) [reply stitcher/src/stitcher.cpp]: `std::move(const&)` typically won't cause a compiler error. It also won't move the object but might create a copy. It's not as performant for many cases but it should be safe. In cases where it's unsafe (eg, making a copy is unsafe) then the the const-copy ctor and const-copy assignment operators should be deleted (for example: std::unique_ptr).

Your code here is fine.

Comment (bluecamel) [reply stitcher/src/stitcher.cpp]: Thanks, @keith-bennett-airmap!  That helps a lot.
